src/components/Trip/TripFullPage/TripFullPage.jsx

```diff
@@ -1,15 +1,126 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import styles from './TripFullPage.module.css';
 import { useParams } from 'react-router-dom';
+import { getUserData } from '../../../utils/getUserData';
+import { doc, getDoc } from 'firebase/firestore';
+import { db } from '../../../config/firebase';
+import { toast } from 'react-hot-toast';
 
 const TripFullPage = () => {
 	const { tripID } = useParams();
 
+	const [trip, setTrip] = useState();
+	const [participantsDataArr, setParticipantsDataArr] = useState();
+
+	const getTrip = async () => {
+		const tripRef = await doc(db, 'Trips', tripID);
+
+		return await getDoc(tripRef).then((tripSnap) => {
```

Comment:
zwrotkę z getDoc raczej przypisałbym do jakiejś zmiennej i sam raczej wrzuciłbym to w try-catch blok. Sposób w jaki napisałaś ten kod zakłada, że nie wystąpi żaden błąd czy to w komunikacji z serwerem czy w zapytaniu samym w sobie np (właśnie wspomniany brak "catch"). Użycie try-catch bloku spowoduje też brak potencjalnej konieczności łączenia kolejnych instrukcji przy pomocy kolejnych .then lub .catch - będzie po prostu czyściej w kodzie ✌️